src/components/c-google-maps.vue

```diff
@@ -0,0 +1,372 @@
+    data() {
+      return {
+        /**
+         * @type {Object} Holds the related Google Maps instance.
+         */
+        mapInstance: null,
+
+        /**
+         * @type {Array.<Object>} Holds a list of map markers.
+         */
+        markers: [],
+
+        /**
+         * @type {Number} Holds the debounce timeout for the bounding.
+         */
+        boundingTimeout: null,
+
+        /**
+         * @type {Boolean} Determines if the component is allowed to auto update the bounding.
+         */
+        allowAutoUpdates: true,
+      };
+    },
+
+    computed: {
+      ...mapGetters('session', [
+        'getGoogleMapsApiKey',
+      ]),
+
+      /**
+       * Maps the locations to the Google Maps required format.
+       *
+       * @returns {Array.<Object>}
+       */
+      mappedLocations() {
+        return this.locations?.map(location => ({
+          lat: parseFloat(location.lat),
+          lng: parseFloat(location.lng),
+          icon: location.icon,
+          geocode: location.geocode,
+          title: location.title,
+          referer: location, // Keep a reference to the original for event payloads.
+        })) || [];
+      },
+
+      /**
+       * Returns a style definition based on the given theme.
+       *
+       * @returns {Array.<Object>}
+       */
+      mapStyles() {
+        switch (this.theme) {
+          case 'gray':
+            return [...GOOGLE_MAPS_THEME_GRAY, ...this.styles];
+
+          default:
+            return this.styles;
+        }
+      },
+    },
+    // watch: {},
+
+    beforeCreate() {
+      const apiKey = store.getters['session/getGoogleMapsApiKey'];
+
+      if (!apiKey) {
+        throw new Error('No Google Maps API key provided.');
+      }
+
+      const url = `https://maps.googleapis.com/maps/api/js?key=${apiKey}&libraries=places`;
+
+      /**
+       * Callback for the load event of the map script.
+       */
+      const callback = () => {
+        geocoder = new window.google.maps.Geocoder();
+
+        this.$nextTick(() => {
+          this.createMapInstance();
+        });
+      };
+
+      loadScript(url, callback, { async: true });
+    },
+    // created() {},
+    // beforeMount() {},
+    mounted() {
+      this.$el.addEventListener('mousedown', this.disableAutoUpdates, { once: true, passive: true });
+    },
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    beforeDestroy() {
+      this.$el.removeEventListener('mousedown', this.disableAutoUpdates);
+    },
+    // destroyed() {},
+
+    methods: {
+      /**
+       * Updates the 'allowAutoUpdates' flag.
+       */
+      disableAutoUpdates() {
+        this.allowAutoUpdates = false;
+      },
+
+      /**
+       * Initialize the google maps with the default values.
+       *
+       * TODO: note, that Google does not support destroying an instance... @see https://stackoverflow.com/questions/10485582/what-is-the-proper-way-to-destroy-a-map-instance
```

Comment:
suggestion: it's unlikely we will ever be able to fix this, so we just make a note out of it.

```suggestion
       * Note: Google does not support destroying a map instance... @see https://stackoverflow.com/questions/10485582/what-is-the-proper-way-to-destroy-a-map-instance
```

Reply:
@patric-eberle One option to destroy a map instance could be is to call the map.setMap(null) method on the map object, and also google.maps.event.clearInstanceListeners(map) to remove all event listeners on the map.

Reply:
@BeneRichi by how it sounds in the video, where the developers give this answer, it does not sound like this would really help to get rid of the instance, only of the reference to the DOM. The only solution seems to be to reuse the instance with other DOM elements, if it is not needed anymore. But that sounds complicated and most likely will never be really a case for us. So I think we can ignore this for now.